### ISSUE TITLE
Can't get it working

### ISSUE THREAD

**Issue by misaka00251** [porelli/firefox-sync#1]
I setup with nginx reverse proxy, then edit `about:config` and sign in my firefox account. The login was successful, but I'm unable to sync.

My `about:sync-log` (I click `Sync now` multiple times):

```
1724489960251	Sync.LogManager	DEBUG	Flushing file log
1724489960256	Sync.LogManager	DEBUG	Log cleanup threshold time: 1723625960256
1724489960258	Sync.LogManager	DEBUG	Done deleting files.
1724489961307	Services.Common.RESTRequest	DEBUG	GET https://profile.accounts.firefox.com/v1/profile 304
1724489963239	Sync.Service	DEBUG	User-Agent: Firefox/130.0 (Windows NT 10.0; Win64; x64) FxSync/1.132.0.20240823112003.desktop
1724489963239	Sync.Service	INFO	Starting sync at 2024-08-24 16:59:23 in browser session nOfZJtMJqg2d
1724489963239	Sync.Service	DEBUG	In sync: should login.
1724489963239	Sync.Service	INFO	User logged in successfully - verifying login.
1724489963239	FirefoxAccounts	TRACE	not checking freshness of profile as it remains recent
1724489963239	Sync.SyncAuthManager	DEBUG	unlockAndVerifyAuthState already has (or can fetch) sync keys
1724489963239	Sync.Status	DEBUG	Status.login: error.login.reason.network => success.status_ok
1724489963239	Sync.Status	DEBUG	Status.service: error.login.failed => error.login.failed
1724489963239	Sync.Service	DEBUG	Fetching unlocked auth state returned success.status_ok
1724489963239	FirefoxAccounts	TRACE	not checking freshness of profile as it remains recent
1724489963240	Sync.Service	DEBUG	verifyLogin failed: TypeError: can't access property "port", uri is null(resource://services-crypto/utils.sys.mjs:437:9) JS Stack trace: computeHAWK@utils.sys.mjs:437:9
_getAuthenticationHeader@sync_auth.sys.mjs:581:41
1724489963240	Sync.Status	DEBUG	Status.login: success.status_ok => error.login.reason.network
1724489963240	Sync.Status	DEBUG	Status.service: error.login.failed => error.login.failed
1724489963240	Sync.ErrorHandler	ERROR	Sync encountered a login error
1724489963240	Sync.SyncScheduler	DEBUG	Clearing sync triggers and the global score.
1724489963240	Sync.SyncScheduler	DEBUG	Next sync in 3600000 ms. (why=schedule)
1724489963240	Sync.Service	DEBUG	Exception calling WrappedLock: Error: Login failed: error.login.reason.network(resource://services-sync/service.sys.mjs:1041:15) JS Stack trace: onNotify@service.sys.mjs:1041:15
1724489963240	Sync.Service	DEBUG	Not syncing: login returned false.
1724489963240	FirefoxAccounts	TRACE	not checking freshness of profile as it remains recent
```

**Comment by misaka00251** [comment]
Something is strange tho after I restart firefox:

```
1724492639861	Services.Common.RESTRequest	DEBUG	GET https://sync.example.com/1.0/sync/1.5 200
1724492639861	Services.Common.TokenServerClient	DEBUG	Got token response: 200
1724492639861	Services.Common.TokenServerClient	DEBUG	Successful token response
1724492639861	Sync.BulkKeyBundle	INFO	BulkKeyBundle being created for undefined
1724492639861	Sync.Status	DEBUG	Status.login: success.status_ok => success.login
1724492639861	Sync.Status	DEBUG	Status.service: success.status_ok => success.status_ok
1724492639862	Sync.SyncAuthManager	DEBUG	_findCluster returning /1.5/4/
1724492639862	Sync.SyncAuthManager	DEBUG	Cluster value = /1.5/4/
1724492639862	Sync.SyncAuthManager	DEBUG	Setting cluster to /1.5/4/
1724492639862	Sync.Service	DEBUG	Caching URLs under storage user base: /1.5/4/
1724492639862	FirefoxAccounts	TRACE	not checking freshness of profile as it remains recent
1724492639862	FirefoxAccounts	TRACE	not checking freshness of profile as it remains recent
1724492639863	Sync.Service	DEBUG	verifyLogin failed: TypeError: can't access property "port", uri is null(resource://services-crypto/utils.sys.mjs:437:9) JS Stack trace: computeHAWK@utils.sys.mjs:437:9
_getAuthenticationHeader@sync_auth.sys.mjs:581:41
```

**Comment by porelli** [comment]
I made a mistake in the docker-compose and forgot to add the `DOMAIN` variable. I tested again with a fresh deployment and now it should work correctly.
Before re-starting the compose, please make sure to:
- update your docker-compose.yml file with the most recent content
- delete your `syncstorage-db` and `tokenserver-db` volumes

Fixed by ac200e3c6be0da43bb4486787dca258e9c8e1ec0
